planner/core/rule_join_reorder_dp.go

```diff
@@ -122,11 +170,12 @@ func (s *joinReorderDPSolver) dpGraph(newPos2OldPos, oldPos2NewPos []int, joinGr
 				continue
 			}
 			// Get the edge connecting the two parts.
-			usedEdges := s.nodesAreConnected(sub, remain, oldPos2NewPos, totalEdges)
+			usedEdges, otherConds := s.nodesAreConnected(sub, remain, oldPos2NewPos, totalEqEdges, totalNonEqEdges)
+			// Here we only check equal condition currently.
 			if len(usedEdges) == 0 {
```

Comment:
If `usedEdges` is empty but `otherConds` is not empty, why not join them like `s.newJoinWithEdge(bestPlan[sub], bestPlan[remain], nil, otherConds)`.
Can we guarantee there must be a better plan than it?

Reply:
1. Here we could only consider the equal conditions since it no need to consider it if the involved nodes of an `other cond` is in the same subgraph connected by equal conditions. But we can take `OtherConds` into special consideration when joining the cartesian groups to bushy join.
2. Haven't done this yet is because that the edge which represents an `other cond` may not be a simple edge. e.g. `t1.a+t2.b=t3.c` is a hyperedge connect three point at once. So keep it as a TODO.